lua/pac3/editor/server/wear.lua

```diff
@@ -1,21 +1,35 @@
+local next = next
+local type = type
+local istable = istable
+local IsValid = IsValid
+local tostring = tostring
+
 pace.StreamQueue = pace.StreamQueue or {}
 
-local frame_number = 0
-local last_frame
-local ERROR_COLOR = Color(228, 37, 37)
+local function startQueueTimer()
+	timer.Create("pac_check_stream_queue", 0.1, 0, function()
+		if pace.BusyStreaming then return end
+
+		local item = table.remove(pace.StreamQueue)
+		if not item then return end
 
-local function catchError(err)
-	pac.Message(ERROR_COLOR, 'Error: ', err)
-	pac.Message(debug.traceback())
+		local data = item.data
+		local filter = item.filter
+		local callback = item.callback
+
+		local allowed, reason = pace.SubmitPartNow(data, filter)
+		if isfunction(callback) then
+			callback(allowed, reason)
+		end
+	end)
```

Comment:
This is the new primary queue timer.

It calls the given callback with `SubmitPartNow`'s return values.

Please note: This doesn't catch errors, so errors will be halting (but the timer will still be rescued).

This isn't _that_ different from how it was before, though (It just printed an error with a debug trace).
Also, because use `table.remove`, there's no chance of getting stuck on a bad item.

The only problem here is that an error in `SubmitPartNow` would prevent the callback from being called. If we need the callback to be guaranteed, we can go back to using `xpcall` (or similar). Though, I believe that JIT isn't able to optimize functions run across protected call boundaries.

If that's true, this function would likely benefit a lot from JIT optimizations. That's why I did it - based only on an unproven years-old email from one of the JIT devs 😁
So I'd be fine changing it if anyone feels strongly about it 👍